libminiooni/libminiooni.go

```diff
@@ -235,14 +235,24 @@ func loadFileInputs(opts *Options) {
 		if len(opts.Inputs) != 0 {
 			fatalWithString("inputs can either be supplied through file or command line, but not both")
 		}
-		content, err := ioutil.ReadFile(opts.InputFilePath)
+		info, err := os.Stat(opts.InputFilePath)
+		if os.IsNotExist(err) {
+			fatalOnError(err, "input file does not exist")
+		}
+		if info.IsDir() {
+			fatalWithString("input file is a directory")
+		}
+		file, err := os.Open(opts.InputFilePath)
```

Comment:
We don't want to be this precise here and only here. There are many places in the codebase where we do not check whether this specific error condition occurs and I don't think they should be neglected.

The code you have written here should be put into a convenience function inside another package. I believe that, for example, it would make sense to create an `internal/fsx` package with file system extensions. I am not suggesting to use `osx` because it would be confusing. I also know Go is planning on introducing `fs` in the standard library that will abstract over the filesystem, hence the name I am proposing.

I think it's fine to introduce this new functionality into the current PR, to do less bureaucratic work, as long as we also mention this new functionality into the title of the PR. (While there, I think we already deal with EOL in the input file: what we're doing here is perhaps fix the way in which we do that?)

I would also be grateful if you could write a test for the new functionality (the name could be `Open`) where we verify that an error is returned in all the cases we care about. For this purpose, we can use a `testdata` directory inside `interal/fsx` where the proper structure of files and directories is pre-added and committed. (We can use a `.gitignore` file containing `*` to force git to keep an otherwise empty directory.)

When these changes are applied, then:

1. we can create an issue about fixing all the other places in which we `os.Open` a file

2. we can apply the suggested diff here and simplify the codebase

```suggestion
		file, err := fsx.Open(opts.InputFilePath)
```

Reply:
I agree.

> 1. we can create an issue about fixing all the other places in which we os.Open a file

Other places are to save the file, not to open it. So I also should write a code to save the file?(To have a complete `fsx`?)

Reply:
(Also: I think I'm not good at naming. :smile:  )